### ISSUE TITLE
Extend MOOSE_MPI_COMMAND to other test objects

### ISSUE THREAD

**Issue by tophmatthews** [idaholab/moose#11751]
## Rationale

Some cluster setups require the use of `srun` or special `mpirun` to handle executions. `MOOSE_MPI_COMMAND` was implemented way back when to handle this for `RunApp` when back when (https://github.com/idaholab/moose/issues/2676). It should be extended to `test_json.py`, `analyzejacobian.py`, and `bench.py`.

## Description

If `srun` is not run on some machines, moose cannot execute. `MOOSE_MPI_COMMAND` fixes this for `RunApp`, but not all testers.

## Impact

Makes the TestHarness applicable to more machines.

**Comment by permcody** [comment]
Tag @milljm

**Comment by permcody** [comment]
I'm not sure that the PostCommand stuff needs to worry about the mpi command. We already run "exodiff" after many of our tests and don't need to do anything special. Are you saying that for slurm, we always need to worry about srun?

**Comment by milljm** [comment]
Looks like I was tagged, but never looked into this... sorry!

I'll try and address this... I also want to link in an older googlegroups thread. Its a bit lengthy, and I think we narrowed it down to a rather nasty issue I am not sure the TestHarness is prepared to solve:

[Installation on Linux Clusters, wanting to use Phase Field module](https://groups.google.com/forum/#!searchin/moose-users/Installation$20on$20Linux$20Clusters$2C$20wanting$20to$20use$20Phase$20Field$20module%7Csort:date/moose-users/tnLwKx_n7PQ/Km50SMn5AwAJ)

Long story short, we found that:

```bash
srun --nodes 1 --mincpus 4 -p branicio -t 2:00:00 mpiexec -n 2 /bin/false
echo $?
```

returns 0

So asking the TestHarness to `srun` any application is not going to be 100% accurate. Everything passes. The tests that found this, were none-other than the tests the TestHarness knows is supposed to fail (way to go RunException).

Maybe you can verify if perhaps, that email thread is an isolated incident and the above srun command indeed returns a non-zero?

**Comment by jessecarterMOOSE** [comment]
@milljm I just ran into this issue on our cluster. FWIW I'm trying to run with `MOOSE_MPI_COMMAND=mpiexec`. For me, not including this causes the application to hang. So any test that launches a process which in turns runs MOOSE times out.

**Comment by milljm** [comment]
Going to link #27561, as I think this PR, once merged, will fix issues involving Slurm :)

Please... if you have the patience, would you try @loganharbour's branch and see if that works?

**Comment by jessecarterMOOSE** [comment]
I don't know if this is a SLURM thing, but I can give it a shot

**Comment by jessecarterMOOSE** [comment]
That branch doesn't appear to help. Can't tell what's being done differently, or if mpiexec is being used to run jobs inside of python scripts, which is what I need.

**Comment by milljm** [comment]
My hope was that users no longer need to set anything, in order allow the TestHarness to interface with your HPC Slurm scheduler. Basically, if you would `unset MOOSE_MPI_COMMAND`, and simply run `./run_test` while on your headnode... I am hopeful that is all that is required.

However, the PR is still a WIP...

**Comment by loganharbour** [comment]
@jessecarterMOOSE glad to offer support if this is something you're getting into. I plan on having the new HPC scheduler merged this month.

**Comment by jessecarterMOOSE** [comment]
@loganharbour HPC support would definitely be appreciated.

**Comment by tophmatthews** [comment]
oh no! Is #30420 the problem? That one can be rolled back, it only helped on a couple tests.

**Comment by jessecarterMOOSE** [comment]
@tophmatthews no I don't think that's the issue. MOOSE_MPI_COMMAND needs to be used in more places I think

**Comment by tophmatthews** [comment]
@jessecarterMOOSE whoops your right! it's been just sitting there so long I assumed it got pushed!
